src/main/java/nextstep/subway/path/domain/SubwayMap.java

```diff
@@ -0,0 +1,101 @@
+package nextstep.subway.path.domain;
+
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import nextstep.subway.line.domain.Line;
+import nextstep.subway.path.exception.PathNotFoundException;
+import nextstep.subway.path.exception.StationNotInGivenLinesException;
+import nextstep.subway.section.domain.Section;
+import nextstep.subway.section.domain.Sections;
+import nextstep.subway.station.domain.Station;
+import org.jgrapht.GraphPath;
+import org.jgrapht.alg.shortestpath.DijkstraShortestPath;
+import org.jgrapht.graph.SimpleDirectedWeightedGraph;
+
+public class SubwayMap {
+    private final List<Line> lines;
+    private final PathType pathType;
+
+    public SubwayMap(List<Line> lines, String type) {
+        this.lines = lines;
+        this.pathType = PathType.of(type);
+    }
+
+    public Path findPath(Station source, Station target) {
+        validateStationsInLines(source, target);
+
+        // 그래프 만들기
+        SimpleDirectedWeightedGraph<Station, SectionEdge> graph = createGraph();
+
+        // 다익스트라 최단 경로 찾기
+        DijkstraShortestPath<Station, SectionEdge> dijkstraShortestPath = new DijkstraShortestPath<>(graph);
+        GraphPath<Station, SectionEdge> result = dijkstraShortestPath.getPath(source, target);
+        validateGraphPath(result);
+
+        List<Section> sections = result.getEdgeList().stream()
+                .map(it -> it.getSection())
```

Comment:
```suggestion
.map(SectionEdge::getSection)
```

개인적으론 이게 더 가독성이 좋더라구요 👀